cns/errorcodeframework/codes.go

```diff
@@ -0,0 +1,141 @@
+package errorcodeframework
+
+type ResponseCode int
+
+// ResponseCode definitions
+// 4000 codes cannot be retried
+// 5000 codes can be retried
+// x100 codes are related to network, connectivity, and environment
+// x300 codes are related to container, orchestrator, and virtualization
+// x500 codes are related to authorization and unexpected errors
+// x700 codes are related agent and internal server errors
+const (
+	Success                                ResponseCode = 2000
+	NetworkContainerVfpProgramComplete     ResponseCode = 2352
+	InvalidParameter                       ResponseCode = 4111
+	MalformedSubnet                        ResponseCode = 4121
+	NetworkContainerNotSpecified           ResponseCode = 4131
+	ReservationNotFound                    ResponseCode = 4141
+	UnknownContainerID                     ResponseCode = 4151
+	UnreachableDockerDaemon                ResponseCode = 4161
+	UnreachableHost                        ResponseCode = 4164
+	UnspecifiedNetworkName                 ResponseCode = 4175
+	UnsupportedEnvironment                 ResponseCode = 4181
+	UnsupportedNetworkType                 ResponseCode = 4185
+	DockerContainerNotSpecified            ResponseCode = 4311
+	EmptyOrchestratorContext               ResponseCode = 4321
+	InvalidPrimaryIPConfig                 ResponseCode = 4341
+	InvalidRequest                         ResponseCode = 4342
+	InvalidSecondaryIPConfig               ResponseCode = 4345
+	NilEndpointStateStore                  ResponseCode = 4356
+	PrimaryCANotSame                       ResponseCode = 4361
+	UnsupportedNCVersion                   ResponseCode = 4381
+	UnsupportedNetworkContainerType        ResponseCode = 4383
+	UnsupportedOrchestratorContext         ResponseCode = 4385
+	UnsupportedOrchestratorType            ResponseCode = 4386
+	UnsupportedVerb                        ResponseCode = 4390
+	StatusUnauthorized                     ResponseCode = 4561
+	UnexpectedError                        ResponseCode = 4580
+	UnsupportedAPI                         ResponseCode = 4591
+	AddressUnavailable                     ResponseCode = 5101
+	CallToHostFailed                       ResponseCode = 5105
+	InconsistentIPConfigState              ResponseCode = 5110
+	NetworkContainerPublishFailed          ResponseCode = 5133
+	NetworkContainerUnpublishFailed        ResponseCode = 5135
+	NetworkJoinFailed                      ResponseCode = 5138
+	NotFound                               ResponseCode = 5140
```

Comment:
What is the object NotFound error here? Is it ServerNotFound or NC Not Found Error?